language-server/docs/CompilerPluginCodeActionExtensions.md

```diff
@@ -0,0 +1,359 @@
+# Language Server Code Action Extension via Compiler Plugins
+
+## Overview
+
+The main goal is to allow ballerina packages to write their own code actions. Mainly the standard libraries have this
+requirement. With this introduction, standard libraries have the capability to show necessary code actions where
+required and apply desired changes once the user click them.
+
+A code action operates in 2 stages:
+
+1. Showing the codeaction - This is when the users see the "bulb" icon in VSCode. It will have a title like
+   `Create variable`, `Create function` or `Document this`.
+
+2. Executing the codeaction - This is when the user clicks on the title mentioned above. Changes (to the source code)
+   required to achieve the task mentioned in the "title" in previous stage should be done here.
```

Comment:
There's no need to mention about command execution since compiler plugin developers need not to know about LSP command execution. Therefore, let's not mention it here.